### PR TITLE
fix: handle foreign key toward non primary key

### PR DESCRIPTION
**Pull request by GuillaumeCisco**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [x] Create automatic tests
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

File: src/services/resource-creator.js [hunk 1]
```diff
@@ -8,18 +8,34 @@ const CompositeKeysManager = require('./composite-keys-manager');
...
-      _.forOwn(model.associations, (association, name) => {
-        if (association.associationType === 'BelongsTo') {
-          promises.push(recordCreated[`set${_.upperFirst(name)}`](params[name], { save: false }));
+    if (associations) {
+      // eslint-disable-next-line no-restricted-syntax
+      for (const [key, value] of Object.entries(associations)) {
+        if (value.associationType === 'BelongsTo') {
+          const pk = params[key];
+          let targetKey = pk;
+          if (value.targetKey !== 'id') { // should we add _id, uuid?
```
Comment (GuillaumeCisco) [inline src/services/resource-creator.js:20]: help is needed here : `should we add _id, uuid?`

File: src/services/belongs-to-updater.js [hunk 2]
```diff
@@ -32,7 +34,15 @@ function BelongsToUpdater(model, assoc, opts, params, data) {
               record[setterName](recordAssociated, options);
             });
         }
-        return record[setterName](data.data ? data.data.id : null, options);
+
+        const pk = data.data ? data.data.id : null;
+        let targetKey = pk;
+        if (pk && modelAssociationTargetKey !== 'id') { // should we add _id, uuid?
```
Comment (GuillaumeCisco) [inline src/services/belongs-to-updater.js:40]: help is needed here : `should we add _id, uuid?`

File: test/databases.test.js [hunk 3]
```diff
@@ -99,11 +100,34 @@ const HasManyDissociator = require('../src/services/has-many-dissociator');
       fieldBad: { type: Sequelize.REAL }, // NOTICE: not supported yet.
     });
 
+    models.owner = sequelize.define('owner', {
+      name: { type: Sequelize.STRING },
+      ownerId: { type: Sequelize.INTEGER, allowNull: false, unique: true },
+    });
+
+    models.project = sequelize.define('project', {
+      name: { type: Sequelize.STRING },
```
Comment (GuillaumeCisco) [inline test/databases.test.js:109]: ownerId is missing

File: src/services/belongs-to-updater.js [hunk 4]
```diff
@@ -24,15 +11,34 @@ class BelongsToUpdater {
     this.data = data;
   }
 
+  // WORKAROUND: Make the hasOne associations update work while waiting
+  //             for the Sequelize 4 release with the fix of the following
+  //             issue: https://github.com/sequelize/sequelize/issues/6069
```
Comment (GuillaumeCisco) [inline src/services/belongs-to-updater.js:16]: This will be done in another pr.
```
if (association.associationType === 'HasOne') {
        targetKey = pk;
      }
```

**Review by guillaumedeslandes — APPROVED** [review]
